Add instructions how to use nf-core 'branding' when only using parts of nf-core template/code etc (#821)
nf-core/nf-co.re: markdown/developers/tutorials/unofficial_pipelines.md

subtitle: Guidance on how to use nf-core code and best practices in non-nf-core pipelines.
---

While we highly recommend and promote contributing to existing nf-core pipelines, in some cases this may not be possible or appropriate. In this case we hope that it will be helpful to use _parts_ of nf-core best practices for your own non-nf-core pipeline. This tutorial describes how to use nf-core code without making official nf-core pipelines.

## Acknowledging nf-core

If you use any form of nf-core code or infrastructure, we kindly ask that you acknowledge the nf-core initiative by adding the following markdown to your `README.md` and pipeline documentation.

```markdown
This pipeline uses code and infrastructure developed and maintained by the [nf-core](https://nf-co.re) community, reused here under the [MIT license](https://github.com/nf-core/tools/blob/master/LICENSE).

> The nf-core framework for community-curated bioinformatics pipelines.
>
> Philip Ewels, Alexander Peltzer, Sven Fillinger, Harshil Patel, Johannes Alneberg, Andreas Wilm, Maxime Ulysse Garcia, Paolo Di Tommaso & Sven Nahnsen.
>
> Nat Biotechnol. 2020 Feb 13. doi: 10.1038/s41587-020-0439-x.

> >

Reviewer: ```suggestion

```

In addition, references of tools and data used in this pipeline are as follows:
```

## Removing nf-core only branding

> Most of the following suggestions refer to when you have started your pipeline using the nf-core template.

## General

- Replace all references to the pipeline name that include the `nf-core/` suffix.

>   - :warning: If you still wish to use the `nf-core tools` linting functionality, this may result in a lot of linting failures. You can create a `.nf-core-lint.yml` file that allows you to ignore or skip certain lint tests. You can find more information on how to do this [here](/tools/#linting-config). One example is the `pipeline_name_conventions`

Reviewer: ```suggestion
  - :warning: If you still wish to use the `nf-core tools` linting functionality, this may result in a lot of linting failures. You can create a `.nf-core.yml` file that allows you to ignore or skip certain lint tests. You can find more information on how to do this [here](/tools/#linting-config). One example is the `pipeline_name_conventions`
```

### `README`

> To ensure users do not get confused, we suggest removing the following components of the main pipeline README

Reviewer: ```suggestion
To ensure users do not get confused, we suggest removing the following components of the main pipeline `README`:
```

1. Remove (or replace) the nf-core logo (see [below](#docs))
2. Slack badge for help messages.
3. Replace all references of the [https://nf-co.re](https://nf-co.re) website to the location of your own documentation
4. Remove reference to the nf-core Slack workspace under `Contributions and Support
5. Replace the `Citations` section with the following markdown:

    ```markdown
    This pipeline uses code and infrastructure developed and maintained by the [nf-core](https://nf-co.re) initative, and reused here under the [MIT license](https://github.com/nf-core/tools/blob/master/LICENSE).

    > The nf-core framework for community-curated bioinformatics pipelines.
    >
    > Philip Ewels, Alexander Peltzer, Sven Fillinger, Harshil Patel, Johannes Alneberg, Andreas Wilm, Maxime Ulysse Garcia, Paolo Di Tommaso & Sven Nahnsen.
    >
    > Nat Biotechnol. 2020 Feb 13. doi: 10.1038/s41587-020-0439-x.
    >

>     In addition, references of tools and data used in this pipeline are as follows:

Reviewer: ```suggestion
    
    In addition, references of tools and data used in this pipeline are as follows:
```

2. `docs/usage.md`: remove reference to documentation being viewable on the [nf-co.re](https://nf-co.re) website
3. `docs/usage.md`: remove reference to the nf-core Slack workspace

Optional:

1. Reference to nf-core/configs in `-profile`
     - Note in many cases it is still reasonable to use these configs for non-nf-core pipelines as they typically only modify generic Nextflow options

### `.github/`

1. `PULL_REQUEST_TEMPLATE.md`: Remove reference to the nf-core/tests-datasets repository
2. `CONTRIBUTING.md`: remove references to the nf-core Slack workspace
3. `workflows/`: remove nf-core specific GitHub actions YAMLs
    - `push_dockerhub*`
4. `ci.yml`: rename to the name of your pipeline rather than `nf-core CI tests`
5. `ISSUE_TEMPLATE/` Remove references to nf-core website in all templates and config files

Optional:

1. `CONTRIBUTING.md`: remove references to the nf-core/tools linting tests
2. `workflows/`: remove the nf-core linting tests under (Note: you can alternatively keep linting and [configure](#general) to ignore certain tests)
    - `linting.yml`
    - `linting_comment.yml`
3. `workflows/`: remove the AWS test GitHub action workflows

> ## Using external module repositories
> 
> <!-- TODO -->

Reviewer: ```suggestion
<!-- TODO: Using external module repositories -->
```

---

> title: "Tutorial: Reusing nf-core components in non-nf-core pipelines. "

Reviewer: Might be a bit on the long side?

Author: Suggestion on how to make it shorter? I went through 3-4 versions and was never happy...

Author: `Using components nf-core in your own pipeline`?

Reviewer 2: `Using nf-core components in your own pipeline` +1
or `Using nf-core components outside nf-core`

Author: Ooh I like the last one

subtitle: Guidance on how to use nf-core code and best practices in non-nf-core pipelines.
---

> While we highly recommend and promote contributing to existing nf-core pipelines, in some cases you may disagree with some of the design decisions, or only wish to follow _parts_ of nf-core best practices for your own non-nf-core pipeline. This tutorial describes how to use nf-core code without making official nf-core pipelines.

Reviewer: ```suggestion
While we highly recommend and promote contributing to existing nf-core pipelines, in some cases this may not be possible or appropriate. In this case we hope that it will be helpful to use _parts_ of nf-core best practices for your own non-nf-core pipeline. This tutorial describes how to use nf-core code without making official nf-core pipelines.
```

## Acknowledging nf-core

> If you use any form of nf-core code or infrastructure, we kindly ask that you acknowledge the nf-core initiative by adding the following markdown to your README or pipeline documentation.

Reviewer: ```suggestion
If you use any form of nf-core code or infrastructure, we kindly ask that you acknowledge the nf-core initiative by adding the following markdown to your `README.md` and pipeline documentation.
```

```markdown

> This pipeline uses code and infrastructure developed and maintained by the [nf-core](https://nf-co.re) initiative, and reused here under the [MIT license](https://github.com/nf-core/tools/blob/master/LICENSE).

Reviewer: ```suggestion
This pipeline uses code and infrastructure developed and maintained by the [nf-core](https://nf-co.re) community, reused here under the [MIT license](https://github.com/nf-core/tools/blob/master/LICENSE).
```

> The nf-core framework for community-curated bioinformatics pipelines.
>
> Philip Ewels, Alexander Peltzer, Sven Fillinger, Harshil Patel, Johannes Alneberg, Andreas Wilm, Maxime Ulysse Garcia, Paolo Di Tommaso & Sven Nahnsen.
>
> Nat Biotechnol. 2020 Feb 13. doi: 10.1038/s41587-020-0439-x.
>
In addition, references of tools and data used in this pipeline are as follows:
```

## Removing nf-core only branding

> Most of the following suggestions refer to when you have started your pipeline using the nf-core template.

## General

> - Replace in all places reference to the pipeline name that includes the `nf-core/` suffix.

Reviewer: ```suggestion
- Replace all references to the pipeline name that include the `nf-core/` suffix.
```

Reviewer: I think some linting tests will fail after doing this. Maybe worth giving an example lint config to get them to pass again.

As you said on slack - a global find and replace command would be good too.

### `README`

> To ensure users do not get confused, we suggest removing the following components of the main pipeline README

Reviewer: Also the main logo! 😃

    >
    > Philip Ewels, Alexander Peltzer, Sven Fillinger, Harshil Patel, Johannes Alneberg, Andreas Wilm, Maxime Ulysse Garcia, Paolo Di Tommaso & Sven Nahnsen.
    >
    > Nat Biotechnol. 2020 Feb 13. doi: 10.1038/s41587-020-0439-x.
    >
    In addition, references of tools and data used in this pipeline are as follows:
    ```

Optional:

1. Reference to nf-core/configs in `Quick Start`
     - Note in many cases it is still reasonable to use these configs for non-nf-core pipelines as they typically only modify generic Nextflow options

### `docs/`

1. Replace the nf-core logo under `docs/images` with your own one.
     - You will also have to change the filename/path at the top the main repository README
2. `docs/usage.md`: remove reference to documentation being viewable on the [nf-co.re](https://nf-co.re) website
3. `docs/usage.md`: remove reference to the nf-core Slack workspace

Optional:

1. Reference to nf-core/configs in `-profile`
     - Note in many cases it is still reasonable to use these configs for non-nf-core pipelines as they typically only modify generic Nextflow options

### `assets/`

> 1. `*email_template.*` - remove references to `nfcorepipelinelogo`

Reviewer: Though if swapped for a different image then maybe this doesn't matter so much..?

Author: I'm not actually sure what the `cid` refers to, so thought it safer to remove :grimacing: 

https://github.com/nf-core/tools/blob/7721f3742fc30a9e6e46d879404bd1366345092d/nf_core/pipeline-template/assets/email_template.html#L13

Reviewer: It's an arbitrary identifier.. It's needed for the attachment / image but it doesn't matter what it is (not visible anywhere unless you look hard when the email is sent).

Reviewer: So if people remove it then the email logo attachment will break. If they replace it with another string it should be fine. If they just leave it alone it shouldn't really matter too.

---

> title: "Tutorial: Reusing nf-core components in non-nf-core pipelines. "

Author: ```suggestion
title: "Tutorial: Using nf-core components outside nf-core"
```

## Acknowledging nf-core

If you use any form of nf-core code or infrastructure, we kindly ask that you acknowledge the nf-core initiative by adding the following markdown to your `README.md` and pipeline documentation.

```markdown
This pipeline uses code and infrastructure developed and maintained by the [nf-core](https://nf-co.re) community, reused here under the [MIT license](https://github.com/nf-core/tools/blob/master/LICENSE).

> The nf-core framework for community-curated bioinformatics pipelines.
>
> Philip Ewels, Alexander Peltzer, Sven Fillinger, Harshil Patel, Johannes Alneberg, Andreas Wilm, Maxime Ulysse Garcia, Paolo Di Tommaso & Sven Nahnsen.
>
> Nat Biotechnol. 2020 Feb 13. doi: 10.1038/s41587-020-0439-x.
>
In addition, references of tools and data used in this pipeline are as follows:
```

## Removing nf-core only branding

> Most of the following suggestions refer to when you have started your pipeline using the nf-core template.

## General

- Replace all references to the pipeline name that include the `nf-core/` suffix.

>     - :warning: If you still wish to use the `nf-core tools` linting functionality, this may result in a lot of linting failures. You can create a `.nf-core-lint.yml` file that allows you to ignore or skip certain lint tests. You can find more information on how to do this [here](/tools/#linting-config). One example is the `pipeline_name_conventions`

Author: ```suggestion
  - :warning: If you still wish to use the `nf-core tools` linting functionality, this may result in a lot of linting failures. You can create a `.nf-core-lint.yml` file that allows you to ignore or skip certain lint tests. You can find more information on how to do this [here](/tools/#linting-config). One example is the `pipeline_name_conventions`
```

### `docs/`

1. Replace the nf-core logo under `docs/images` with your own one.
     - You will also have to change the filename/path at the top the main repository README
2. `docs/usage.md`: remove reference to documentation being viewable on the [nf-co.re](https://nf-co.re) website
3. `docs/usage.md`: remove reference to the nf-core Slack workspace

Optional:

1. Reference to nf-core/configs in `-profile`
     - Note in many cases it is still reasonable to use these configs for non-nf-core pipelines as they typically only modify generic Nextflow options

### `assets/`

1. `*email_template.*` - remove references to `nfcorepipelinelogo`

### `.github/`

1. `PULL_REQUEST_TEMPLATE.md`: Remove reference to the nf-core/tests-datasets repository
2. `CONTRIBUTING.md`: remove references to the nf-core Slack workspace
3. `workflows/`: remove nf-core specific GitHub actions YAMLs
    - `push_dockerhub*`
4. `ci.yml`: rename to the name of your pipeline rather than `nf-core CI tests`
5. `ISSUE_TEMPLATE/` Remove references to nf-core website in all templates and config files

Optional:

1. `CONTRIBUTING.md`: remove references to the nf-core/tools linting tests

> 2. `workflows/`: remove the nf-core linting tests under

Reviewer 2: you might want to keep the linting, I'd recommend configuring the linting

> 2. `workflows/`: remove the nf-core linting tests under

Author: ```suggestion
2. `workflows/`: remove the nf-core linting tests under (Note: you can alternatively keep linting and [configure](#general) to ignore certain tests)
```
